src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3049,7 +3049,18 @@ protected virtual void ProcessMethod (MethodDefinition method, in DependencyInfo
 				}
 			}
 
+			// Mark overrides except for static interface methods
 			if (method.HasOverrides) {
+				List<MethodReference> staticInterfaceOverrideIndices = new ();
+				foreach (MethodReference ov in method.Overrides) {
+					if (_context!.Resolve (ov)?.IsStatic == true
+						&& _context!.Resolve (ov.DeclaringType)?.IsInterface == true) {
+						staticInterfaceOverrideIndices.Add (ov);
+					}
+				}
+				foreach (var overrideToRemove in staticInterfaceOverrideIndices) {
+					method.Overrides.Remove (overrideToRemove);
```

Comment:
What happens if we decide to keep the method because it's called? It would probably still remain removed from the overrides, right? That sounds problematic. I don't know the exact reasons why the compiler generates the explicit override, but I assume there is a reason - and linker should not go against it.

Reply:
Great point, I changed it to only remove the annotation if the interface method is also removed